### PR TITLE
Feat/dev_lesson_15

### PR DESCRIPTION
**Pull request by dev-benson-03**
### **Overview**
This lesson focuses on Test Driven Development (TDD) principles and implementing comprehensive test coverage for Java applications, along with creating integration tests for TypeScript applications.

###  **Completed Assignments**

- Employee Class: Implemented with full getter/setter methods and formatted output
- EmployeeManager Class: Service for managing employee CRUD operations
- Comprehensive Unit Tests: 100% line coverage achieved
- Code Coverage: Using JaCoCo with 80% minimum threshold (achieved 100%)
- Stretch: Copied Lesson 10 Libraries, Integrated Tests, Code Coverage Fits Framework, Separate Test Scripts: 

### **Key Features:**

- Employee model with id, name, department, salary, and details fields
- EmployeeManager with add, get, update, remove, and count operations
- Exception handling for invalid operations
- Edge case testing for null values, empty strings, and boundary conditions

### **2. Stretch Assignment (TypeScript)** 

- Copied Lesson 10 Libraries: Complete TypeScript project structure
- Integration Tests: Tests for MediaCollectionApp CSV loading functionality
- Code Coverage Framework: Jest with 70% minimum threshold
- Separate Test Scripts: Unit tests and integration tests run independently

### **Test Coverage Results**
_TDD App (Java)_
Current Coverage: 100% (1.000 in decimal format)
Required Minimum: 80% (0.8 in decimal format)
Status: ✅ EXCEEDS REQUIREMENT

_Libraries App (TypeScript)_
Framework: Jest with coverage collection
Required Minimum: 70% (0.7 in decimal format)
Configuration: HTML, LCOV, and text report formats

### **Running Tests**
TDD App (Java)
Libraries App (TypeScript)

### CODE REVIEW

File: lesson_15/stretch_libraries/src/integration/media_collection_app.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { MediaCollectionApp } from '@/cli/media_collection_app';
+import { AppModule } from '@/app.module';
+import { MediaCollection } from '@/models/media_collection';
+import { DeanWalstonLoader } from '@/loaders/dean_walston_loader';
+import { BenjaminScottLoader } from '@/loaders/benjamin_scott_loader';
+import { JaredEdgeLoader } from '@/loaders/jared_edge_loader';
+
+describe('MediaCollectionApp Integration Tests', () => {
```
Comment (anthonydmays) [inline lesson_15/stretch_libraries/src/integration/media_collection_app.integration.test.ts:9]: These aren't really integration tests

File: lesson_15/tdd/tdd_app/src/main/java/com/codedifferently/lesson15/Employee.java [hunk 2]
```diff
@@ -45,4 +46,12 @@ public double getSalary() {
   public void setSalary(double salary) {
     this.salary = salary;
   }
+
+  public String getDetails() {
```
Comment (anthonydmays) [inline lesson_15/tdd/tdd_app/src/main/java/com/codedifferently/lesson15/Employee.java:50]: This doesn't meet the requirements for `getDetails`.
